src/explorerCommands/new.ts

```diff
@@ -90,7 +107,7 @@ function getNewFilePath(basePath: string, className: string): string {
 }
 
 export async function newPackage(node: DataNode): Promise<void> {
-    if (!node.uri) {
+    if (!node?.uri || !canCreatePackage(node)) {
```

Comment:
Just one concern. If no node is selected, by triggering keyboard shortcut, nothing would happen. Is that true? If so, "Creating new packages or classes doesn't work using hotkeys" is still true in such scenarios.

I think it would be better to let users know what to do next. 
* Better experience: Prompt to select target package
* Easy implementation: notification with instructions

Reply:
Good point. Let me think how to improve this